pkg/kubelet/kubelet.go

```diff
@@ -2457,6 +2457,61 @@ func (kl *Kubelet) setNodeStatus(node *api.Node) error {
 			kl.recordNodeStatusEvent("NodeNotReady")
 		}
 	}
+
```

Comment:
IMO, we should refactor and move all node status code out of kubelet.go. This should include factor out common code for setting conditions. The pattern here is hard to follow. That deserves its own PR though.